pyabc/acceptor/acceptor.py

```diff
@@ -0,0 +1,288 @@
+
+        Returns
+        -------
+
+        An AcceptorResult.
+
+        .. note::
+            Currently, only one value encoding the distance is returned
+            (and stored in the database),
+            namely that at time t, even if also other distances affect the
+            acceptance decision, e.g. distances from previous iterations,
+            or in general if the distance is not scalar.
+            This is because the last distance is likely to be most informative
+            for the further process, and because in some parts of ABC a scalar
+            distance value is required.
+        """
+        raise NotImplementedError()
+
+
+class SimpleFunctionAcceptor(Acceptor):
+    """
+    Initialize from function.
+
+    Parameters
+    ----------
+
+    fun: Callable, optional
+        Callable with the same signature as the __call__ method.
+    """
+
+    def __init__(self, fun=None):
+        super().__init__()
+        self.fun = fun
+
+    def __call__(self, distance_function, eps, x, x_0, t, par):
+        return self.fun(distance_function, eps, x, x_0, t, par)
+
+    @staticmethod
+    def assert_acceptor(maybe_acceptor):
+        """
+        Create an acceptor object from input.
+
+        Parameters
+        ----------
+
+        maybe_acceptor: Acceptor or Callable
+            Either pass a full acceptor, or a callable which is then filled
+            into a SimpleAcceptor.
+
+        Returns
+        -------
+
+        acceptor: Acceptor
+            An Acceptor object in either case.
+        """
+        if isinstance(maybe_acceptor, Acceptor):
+            return maybe_acceptor
+        else:
+            return SimpleFunctionAcceptor(maybe_acceptor)
+
+
+def accept_use_current_time(
+        distance_function, eps, x, x_0, t, par):
+    """
+    Use only the distance function and epsilon criterion at the current time
+    point to evaluate whether to accept or reject.
+    """
+
+    d = distance_function(x, x_0, t, par)
+    accept = d <= eps(t)
+
+    return AcceptorResult(distance=d, accept=accept)
+
+
+def accept_use_complete_history(
+        distance_function, eps, x, x_0, t, par):
+    """
+    Use the acceptance criteria from the complete history to evaluate whether
+    to accept or reject.
+
+    This includes time points 0,...,t, as far as these are
+    available. If either the distance function or the epsilon criterion cannot
+    handle any time point in this interval, the resulting error is simply
+    intercepted and the respective time not used for evaluation. This situation
+    can frequently occur when continuing a stopped run. A different behavior
+    is easy to implement.
+    """
+    # first test current criterion, which is most likely to fail
+    d = distance_function(x, x_0, t, par)
+    accept = d <= eps(t)
+
+    if accept:
+        # also check against all previous distances and acceptance criteria
+        for t_prev in range(0, t):
```

Comment:
can be replaced with `range(t)`

Reply:
True, but I like to write things explicitly :)